### PR TITLE
Add Enterprise Search node client docs

### PR DESCRIPTION
**Pull request by leemthompo**
- [Enterprise Search issue ](https://github.com/elastic/enterprise-search-team/issues/3549) (private repo)
- Content was iterated on and approved in [a draft PR](https://github.com/elastic/enterprise-search-pubs/pull/3081) in the Enterprise Search repo.

### CODE REVIEW

**Review by chriscressman — CHANGES_REQUESTED** [review]
I forgot that this repository contains two separate packages. I think the right way to package up the docs for this repo would be to create a book for each package (although we'll only initially publish the Node.js book). Interestingly, the [universal package already has some docs](https://github.com/elastic/enterprise-search-js/tree/main/packages/enterprise-search-universal/docs), while the Node.js package doesn't.

For this PR, which focuses on the Node.js package, I think we should move the docs into the directory for that package (similar to how the "universal" package is structured now). Then we should remove the `node-client-` prefix from filenames and IDs.

File: packages/enterprise-search/docs/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+= enterprise-search-js
+
+:doctype:           book
+
+include::{asciidoc-dir}/../../../shared/attributes.asciidoc[]
```
Comment (leemthompo) [inline packages/enterprise-search/docs/index.asciidoc:5]: @chriscressman I'm uncertain about this variable. Is it nested correctly here?

Comment (chriscressman) [inline packages/enterprise-search/docs/index.asciidoc]: Here's what we have in the main Enterprise Search docs. This is probably the most up to date version of these includes:

```suggestion
include::{docs-root}/shared/versions/stack/{source_branch}.asciidoc[]
include::{docs-root}/shared/attributes.asciidoc[]
```

File: packages/enterprise-search/docs/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+= enterprise-search-js
```
Comment (chriscressman) [inline packages/enterprise-search/docs/index.asciidoc:1]: Adding ID here for consistency with other files:

```suggestion
[#index]
= enterprise-search-js
```

File: packages/enterprise-search/docs/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+:doctype:           book
```
Comment (chriscressman) [inline packages/enterprise-search/docs/index.asciidoc:3]: I don't think this does anything.

```suggestion
```

**Comment by leemthompo** [comment]
- [x] Move the docs into the directory for that package
- [x] Remove the node-client- prefix from filenames and IDs

File: packages/enterprise-search/docs/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+include::installation.asciidoc[]
+
+include::app-search-api.asciidoc[]
+
+include::workplace-search-api.asciidoc[]
+
+include::enterprise-search-api.asciidoc[]
+
+include::release-notes/index.asciidoc[]
```
Comment (chriscressman) [inline packages/enterprise-search/docs/index.asciidoc:17]: I suggest flattening this out (remove directory) to follow the pattern in the main Enterprise Search docs.

**Review by chriscressman — APPROVED** [review]

File: packages/enterprise-search/docs/index.asciidoc (hunk 3 above)
Comment (leemthompo) [reply packages/enterprise-search/docs/index.asciidoc:3]: A Chesterton fence? 😄

**Pull request merged**